PC/winreg.c

```diff
@@ -261,16 +283,22 @@ class self_return_converter(CReturnConverter):
 /*[clinic input]
 winreg.HKEYType.Close
 
+    cls: defining_class
+
```

Comment:
HKEY type is not subclassable, so you can revert the clinic changes and just use `Py_TYPE(self)` and then `_PyType_GetModuleState`. Ditto for `__exit__`.

Reply:
Reverted the changes and just using `Py_TYPE(self)`, but still keep using of `PyType_GetModule` instead of the suggested `_PyType_GetModuleState`.

The reason is we will passing the type instance to other functions `PyHKEY_Close` and `PyHKEY_AsHKEY`, passing a `module` as parameter is more flexible as the module level function may access more context in the module object.

I don't know is this right?

Reply:
Yep, it is flexible, but for example for `winreg_ConnectRegistry`, you end up calling `PyType_GetModule` twice in the clinic code (first in `clinic_HKEY_converter`, then in `PyHKEY_FromHKEY`). Perhaps that does not matter much, wrt. performance.